src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecInputParameterConverter.cs

```
@@ -0,0 +1,124 @@
+                    || reader.TryReadBoolean(nameof(InputParameter.IsRequired), ref isRequired)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsApiVersion), ref isApiVersion)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsResourceParameter), ref isResourceParameter)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsContentType), ref isContentType)
+                    || reader.TryReadBoolean(nameof(InputParameter.IsEndpoint), ref isEndpoint)
+                    || reader.TryReadBoolean(nameof(InputParameter.SkipUrlEncoding), ref skipUrlEncoding)
+                    || reader.TryReadBoolean(nameof(InputParameter.Explode), ref explode)
+                    || reader.TryReadString(nameof(InputParameter.ArraySerializationDelimiter), ref arraySerializationDelimiter)
+                    || reader.TryReadString(nameof(InputParameter.HeaderCollectionPrefix), ref headerCollectionPrefix);
+
+                if (!isKnownProperty)
+                {
+                    reader.SkipProperty();
+                }
+            }
+
+            name = name ?? throw new JsonException("Parameter must have name");
+            nameInRequest = nameInRequest ?? throw new JsonException("Parameter must have nameInRequest");
+            parameterType = parameterType ?? throw new JsonException("Parameter must have type");
+
+            if (location == null)
+            {
+                throw new JsonException("Parameter must have location");
+            }
+            RequestLocation requestLocation;
+            Enum.TryParse<RequestLocation>(location, ignoreCase: true, out requestLocation);
+
+            if (kind == null)
+            {
+                throw new JsonException("Parameter must have kind");
+            }
+            InputOperationParameterKind parameterKind;
+            Enum.TryParse<InputOperationParameterKind>(kind, ignoreCase: true, out parameterKind);
```

How about we combine these two lines or just use Enum.Parse and let it panic when it cannot 
```suggestion
            Enum.TryParse<InputOperationParameterKind>(kind, ignoreCase: true, out var parameterKind);
```